contrib/functions/go/delete-annotations/main.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	annotationKeysLiteral = "annotationKeys"
+	annotationDelimeter   = ","
+)
+
+//nolint
+func main() {
+	fp := DeleteAnnotationsProcessor{}
+	cmd := command.Build(&fp, command.StandaloneEnabled, false)
+	if err := cmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+func (fp *DeleteAnnotationsProcessor) Process(resourceList *framework.ResourceList) error {
+	resourceList.Result = &framework.Result{
+		Name: "delete-annotations",
+	}
+
+	// get the annotation key from functionConfig
+	var annotationKeys string
+	err := getAnnotationKeys(resourceList.FunctionConfig, &annotationKeys)
+	if err != nil {
+		resourceList.Result.Items = getErrorItem(err.Error())
+		return err
+	}
+
+	// process resources in the package and delete the annotation keys provided
+	items, err := processResources(resourceList.Items, annotationKeys)
+	if err != nil {
+		resourceList.Result.Items = getErrorItem(err.Error())
+		return err
+	}
+	resourceList.Result.Items = items
+	return nil
+}
+
+func processResources(resourceList []*yaml.RNode, keys string) ([]framework.ResultItem, error) {
+	var resultItems []framework.ResultItem
+	for _, node := range resourceList {
+		if node.IsNilOrEmpty() {
+			continue
+		}
+
+		//confirm resources are valid
+		metadata, err := node.GetMeta()
+		if err != nil {
+			return nil, err
+		}
+
+		if metadata.Name == "" || metadata.Kind == "" {
+			continue
+		}
+
+		keysList := strings.Split(strings.TrimSpace(keys), annotationDelimeter)
+
+		for _, annotationKey := range keysList {
+			mutatedNode, err := node.Pipe(yaml.Lookup("metadata", "annotations"), yaml.FieldClearer{
```

Comment:
Just curious, how hard do you feel about searching and using functions like `yaml.FieldClearer`?

Reply:
Not fond of it. Piping wasn't a good experience. It's much simple now with all (most) CRUD operations available on `KubeObject`